openmetadata-ui/src/main/resources/ui/src/components/ProfilerDashboard/component/TestSummary.test.tsx

```diff
@@ -0,0 +1,156 @@
+    .fn()
+    .mockImplementation(() => <div>DatePickerMenu.component</div>);
+});
+jest.mock('../../common/error-with-placeholder/ErrorPlaceHolder', () => {
+  return jest
+    .fn()
+    .mockImplementation(() => <div>ErrorPlaceHolder.component</div>);
+});
+jest.mock('../../Loader/Loader', () => {
+  return jest.fn().mockImplementation(() => <div>Loader.component</div>);
+});
+jest.mock('../../schema-editor/SchemaEditor', () => {
+  return jest.fn().mockImplementation(() => <div>SchemaEditor.component</div>);
+});
+
+describe('TestSummary component', () => {
+  it('Component should render', async () => {
+    render(<TestSummary {...mockProps} />);
+
+    const graphContainer = await screen.findByTestId('graph-container');
+    const graph = queryByAttribute(
+      'id',
+      graphContainer,
+      `${mockProps.data.name}_graph`
+    );
+
+    expect(
+      await screen.findByTestId('test-summary-container')
+    ).toBeInTheDocument();
+    expect(
+      await screen.findByTestId('test-case-expand-button')
+    ).toBeInTheDocument();
+    expect(await screen.findByTestId('params-container')).toBeInTheDocument();
+    expect(graphContainer).toBeInTheDocument();
+    expect(graph).toBeInTheDocument();
+    expect(
+      await screen.findByText('DatePickerMenu.component')
+    ).toBeInTheDocument();
+    expect(await screen.findAllByTestId('parameter-value')).toHaveLength(
+      mockProps.data.parameterValues?.length || 0
+    );
+  });
+
+  it('Show no data placeholder when there is no result, other CTA should also visible', async () => {
+    (getListTestCaseResults as jest.Mock).mockImplementationOnce(() =>
+      Promise.resolve({ data: [] })
+    );
+    render(<TestSummary {...mockProps} />);
+    const graphContainer = await screen.findByTestId('graph-container');
+    const graph = queryByAttribute(
+      'id',
+      graphContainer,
+      `${mockProps.data.name}_graph`
+    );
+
+    expect(graphContainer).toBeInTheDocument();
+    expect(graph).not.toBeInTheDocument();
+    expect(
+      await screen.findByText('ErrorPlaceHolder.component')
+    ).toBeInTheDocument();
+    expect(
+      await screen.findByTestId('test-case-expand-button')
+    ).toBeInTheDocument();
+    expect(await screen.findByTestId('params-container')).toBeInTheDocument();
+
+    expect(await screen.findAllByTestId('parameter-value')).toHaveLength(
+      mockProps.data.parameterValues?.length || 0
+    );
+  });
+
+  it('Schema edito should be visible if sqlExpression is available in params', async () => {
+    render(<TestSummary data={MOCK_SQL_TEST_CASE} />);
+
+    expect(
+      await screen.findByText('SchemaEditor.component')
+    ).toBeInTheDocument();
+  });
+
+  it('full screen button click should work', async () => {
+    render(<TestSummary data={MOCK_SQL_TEST_CASE} />);
+
+    const fullScreenBtn = await screen.findByTestId('test-case-expand-button');
+
+    expect(fullScreenBtn).toBeInTheDocument();
+
+    fireEvent.click(fullScreenBtn);
```

Comment:
I would recommend using the `userEvent` here as it provides a more realistic simulation of user interactions and can catch issues that fireEvent might miss.